carapace-server/src/main/java/org/carapaceproxy/server/certificates/ACMEClient.java

```diff
@@ -155,6 +151,29 @@ private Http01Challenge httpChallenge(Authorization auth) throws AcmeException {
         return challenge;
     }
 
+    /**
+     * Prepares a DNS challenge.
+     * <p>
+     * The verification of this challenge expects a TXT record with a certain content.
+     * </p>
+     *
+     * @param auth {@link Authorization} to find the challenge in
+     * @return {@link Challenge} to verify
+     */
+    public Challenge dnsChallenge(Authorization auth) throws AcmeException {
+        // Find a single dns-01 challenge
+        Dns01Challenge challenge = auth.findChallenge(Dns01Challenge.TYPE);
+        if (challenge == null) {
```

Comment:
in this case the system will break
how can the operator get out of this situation ?

Reply:
This happens when the CA does not support DNS challenges. For Let's encrypt will be != null.